src/write-yaml-template.ts

```diff
@@ -0,0 +1,45 @@
+import * as fs from 'fs'
+import * as path from 'path'
+import { stringify } from 'json-to-pretty-yaml'
+
+const walk = (dir: string, prevResults?: string[]): string[] => {
+  const list = fs.readdirSync(dir)
+  const results = prevResults || []
+  list.forEach((file) => {
+    file = path.resolve(dir, file)
+    const stat = fs.statSync(file)
+    if (stat && stat.isDirectory()) {
+      walk(file, results)
+    } else {
+      results.push(file)
+    }
+  })
+  return results
+}
```

Comment:
I sense this would be more testable if I removed the dependency on fs

Reply:
Or you could use one of the many libraries which have this functionality built in. ;)
- [klaw](https://www.npmjs.com/package/klaw) or [klaw-sync](https://www.npmjs.com/package/klaw-sync)
- [fs-extra](https://www.npmjs.com/package/fs-extra)
- [walker](https://www.npmjs.com/package/walker)
- [walk-sync](https://www.npmjs.com/package/walk-sync)